### PR TITLE
Use script_retry instead of shell for loops

### PR DESCRIPTION
**Pull request by kalikiana**
Fixes: https://progress.opensuse.org/issues/110803

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]
First, why should script_retry be better than the bash-internal loop? Second, where should script_retry come from? I think again you are confusing os-autoinst-distri-openQA and os-autoinst-distri-opensuse. In any case I prefer the bash internal loop to prevent unnecessary scope switching between bash within the SUT and perl code on the outside worker environment

**Comment by okurz** [comment]
> It is based on a suggestion by @grisu48 and yes, I realized after pushing the initial branch that this is a function from distri-opensuse. If you think this is a bad approach, maybe it should be re-considered in os-autoinst-distri-opensuse as well?

Maybe but I leave it to the maintainers of os-autoinst-distri-opensuse to decide that. For this test distribution here I see no benefit

**Comment by Martchus** [comment]
> In any case I prefer the bash internal loop to prevent unnecessary scope switching between bash within the SUT and perl code on the outside worker environment

That's actually a good point.

I mainly opted for using the helper because it introduces a delay between the attempts (which might otherwise be pointless anyways). However, a delay can also be added on Bash level.

**Comment by kalikiana** [comment]
Verification run: https://openqa.opensuse.org/tests/2339756

Well, no strong opinion on my side. I figured I'd try what was suggested since it's proven to work well in some cases.

File: lib/utils.pm
Comment (kalikiana) [reply lib/utils.pm:131]: I don't understand your comment given your saying the function worse than what we have.

**Pull request closed without merging**
